client/v3/client.go

```diff
@@ -285,22 +288,34 @@ func (c *Client) Dial(ep string) (*grpc.ClientConn, error) {
 	return c.dial(creds, grpc.WithResolvers(resolver.New(ep)))
 }
 
+// UpdateAuthToken allows updating the JWT auth token held by the
+// client. It is safe to call this function concurrently with other
+// operations.
+func (c *Client) UpdateAuthToken(token string) {
+	c.authTokenBundle.UpdateAuthToken(token)
+}
```

Comment:
I suggest not to add an exported/public method.
```suggestion
```

Reply:
I exposed this publicly because it's the only way to update an auth token directly in the client (since `authTokenBundle` is not visible to consumers of the client).

The use-case for exposing this is that a consumer of the client may be managing token lifecycle entirely outside of etcd and issue a replacement token for an expired token out-of-band from the client and the consumer of the client will need to call `UpdateAuthToken` to replace the expired token with the new one.

Reply:
> Given that use-case can we leave this exported?

OK.